src/cunumeric/unary/unary_red_util.h

```diff
@@ -72,146 +74,192 @@ constexpr decltype(auto) op_dispatch(UnaryRedCode op_code, Functor f, Fnargs&&..
+  __CUDA_HD__ static VAL convert(const Legion::Point<DIM>&, int32_t, const RHS& rhs)
+  {
+    return rhs;
+  }
+
+  __CUDA_HD__ static VAL convert(const RHS& rhs) { return rhs; }
 };
 
 template <legate::LegateTypeCode TYPE_CODE>
 struct UnaryRedOp<UnaryRedCode::MIN, TYPE_CODE> {
-  static constexpr bool valid = true;
+  static constexpr bool valid = TYPE_CODE != legate::LegateTypeCode::COMPLEX128_LT;
 
-  using VAL = legate::legate_type_of<TYPE_CODE>;
+  using RHS = legate::legate_type_of<TYPE_CODE>;
+  using VAL = RHS;
   using OP  = Legion::MinReduction<VAL>;
 
   template <bool EXCLUSIVE>
-  __CUDA_HD__ static void fold(VAL& rhs1, VAL rhs2)
+  __CUDA_HD__ static void fold(VAL& a, VAL b)
   {
-    OP::template fold<EXCLUSIVE>(rhs1, rhs2);
+    OP::template fold<EXCLUSIVE>(a, b);
   }
-};
 
-template <>
-struct UnaryRedOp<UnaryRedCode::MIN, legate::LegateTypeCode::COMPLEX128_LT> {
-  static constexpr bool valid = false;
+  template <int32_t DIM>
+  __CUDA_HD__ static VAL convert(const Legion::Point<DIM>&, int32_t, const RHS& rhs)
+  {
+    return rhs;
+  }
+
+  __CUDA_HD__ static VAL convert(const RHS& rhs) { return rhs; }
 };
 
 template <legate::LegateTypeCode TYPE_CODE>
 struct UnaryRedOp<UnaryRedCode::PROD, TYPE_CODE> {
-  static constexpr bool valid = true;
+  static constexpr bool valid = TYPE_CODE != legate::LegateTypeCode::COMPLEX128_LT;
 
-  using VAL = legate::legate_type_of<TYPE_CODE>;
+  using RHS = legate::legate_type_of<TYPE_CODE>;
+  using VAL = RHS;
   using OP  = Legion::ProdReduction<VAL>;
 
   template <bool EXCLUSIVE>
-  __CUDA_HD__ static void fold(VAL& rhs1, VAL rhs2)
+  __CUDA_HD__ static void fold(VAL& a, VAL b)
   {
-    OP::template fold<EXCLUSIVE>(rhs1, rhs2);
+    OP::template fold<EXCLUSIVE>(a, b);
   }
-};
 
-template <>
-struct UnaryRedOp<UnaryRedCode::PROD, legate::LegateTypeCode::COMPLEX128_LT> {
-  static constexpr bool valid = false;
+  template <int32_t DIM>
+  __CUDA_HD__ static VAL convert(const Legion::Point<DIM>&, int32_t, const RHS& rhs)
+  {
+    return rhs;
+  }
+
+  __CUDA_HD__ static VAL convert(const RHS& rhs) { return rhs; }
 };
 
 template <legate::LegateTypeCode TYPE_CODE>
 struct UnaryRedOp<UnaryRedCode::SUM, TYPE_CODE> {
   static constexpr bool valid = true;
 
-  using VAL = legate::legate_type_of<TYPE_CODE>;
+  using RHS = legate::legate_type_of<TYPE_CODE>;
+  using VAL = RHS;
   using OP  = Legion::SumReduction<VAL>;
 
   template <bool EXCLUSIVE>
-  __CUDA_HD__ static void fold(VAL& rhs1, VAL rhs2)
+  __CUDA_HD__ static void fold(VAL& a, VAL b)
+  {
+    OP::template fold<EXCLUSIVE>(a, b);
+  }
+
+  template <int32_t DIM>
+  __CUDA_HD__ static VAL convert(const Legion::Point<DIM>&, int32_t, const RHS& rhs)
   {
-    OP::template fold<EXCLUSIVE>(rhs1, rhs2);
+    return rhs;
   }
+
+  __CUDA_HD__ static VAL convert(const RHS& rhs) { return rhs; }
 };
 
 template <legate::LegateTypeCode TYPE_CODE>
 struct UnaryRedOp<UnaryRedCode::ARGMAX, TYPE_CODE> {
   static constexpr bool valid = true;
```

Comment:
should you replace here `valid =true` with `static constexpr bool valid = TYPE_CODE != legate::LegateTypeCode::COMPLEX128_LT;`  as you did for other specializations? The same for `ARGMAX`

Reply:
08fd23f7c909db55ed87d1d7a4977bfef2960dfc